### PR TITLE
Enable registering a specific DOI

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: lib/datacite/register_doi.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Datacite
+  # The JSON request to create a specific DOI
+  class RegisterDoi
```
Comment (ndushay) [inline lib/datacite/register_doi.rb:5]: I would expect a class named "RegisterDoi" to do the actual registration.  Perhaps `DoiRequestJSON` ?  or even `DOIRequestJSON`  or just plain `DOIRequest`?

**Review by ndushay — APPROVED** [review]
a naming change to be considered.

File: lib/datacite/register_doi.rb (hunk 1 above)
Comment (jcoyne) [reply lib/datacite/register_doi.rb:5]: I think that "register" has to be in there somewhere, so as not to confuse it with retrieving a DOI.

Comment (jcoyne) [reply lib/datacite/register_doi.rb:5]: I don't like `JSON` in the name because this object is not the JSON, but has a method that produces JSON.

Comment (jcoyne) [reply lib/datacite/register_doi.rb:5]: So... `RegisterDoiRequestBodyFactory`?

Comment (ndushay) [reply lib/datacite/register_doi.rb:5]: I think a request for a DOI is different from retrieving a DOI.   But having a method in one class that is named "registerDOI" ... that actually does the registration, and a class named "RegisterDOI" that does NOT do the registration ... smells.

Comment (ndushay) [reply lib/datacite/register_doi.rb:5]: `RegisterDoiRequestBody` or with the factory is clearer, yes.

**Review by ndushay — APPROVED** [review]

**Pull request merged**